### PR TITLE
Feature/hooks

### PR DESCRIPTION
**Pull request by ghost**
## Summary

* Implement useMemo into the filterInput

## Trello Card

[Trello Card](https://trello.com/c/GX9c8xec/64-n-113-hooks-ii-usememo)

### CODE REVIEW

File: wbooks/src/app/components/FilterInput/index.tsx [hunk 1]
```diff
@@ -15,9 +16,14 @@ function FilterInput() {
...
-  useEffect(() => {
+  useMemo(() => {
     if (debounceSearchTerm) {
-      dispatch(BookActions.getFilteredBooks(value));
+      const filteredBooks = value
+        ? BOOKS_MOCK.filter(book => {
+            return book.title.toLowerCase().includes(value.toLowerCase());
+          })
+        : [];
+      dispatch(BookActions.getFilteredBooks({ books: filteredBooks, query: value }));
```
Comment (sfernandez11) [inline wbooks/src/app/components/FilterInput/index.tsx]: Wrong use of `useMemo` hook

File: wbooks/src/redux/book/actions.ts [hunk 2]
```diff
@@ -12,11 +13,11 @@ const actionCreators = {
     target: 'books',
     service: getBooks
   }),
-  getFilteredBooks: (query: string) => ({
+  getFilteredBooks: (data: FilteredBooks) => ({
     type: actions.GET_FILTERED_BOOKS,
     target: 'filteredBooks',
     service: getFilteredBooks,
-    payload: query
+    payload: data
```
Comment (sfernandez11) [inline wbooks/src/redux/book/actions.ts]: No, you must save the `query` and use the hook/filter in the `Filter` screen only

File: wbooks/src/services/BookService.ts [hunk 3]
```diff
@@ -1,14 +1,6 @@
...
-  if (!query) {
-    return Promise.resolve({ ok: true, data: { books: [], query }, problem: '' });
-  }
-
-  const filteredBooks = BOOKS_MOCK.filter(book =>
-    book.title.toLocaleLowerCase().includes(query.toLocaleLowerCase())
-  );
-  return Promise.resolve({ ok: true, data: { books: filteredBooks, query }, problem: '' });
-};
+export const getFilteredBooks = (data: FilteredBooks) => Promise.resolve({ ok: true, data, problem: '' });
```
Comment (sfernandez11) [inline wbooks/src/services/BookService.ts]: No, you must use the same `getBooks`,  this is wrong and unnecessary

**Review by sfernandez11 — CHANGES_REQUESTED** [review]

File: wbooks/src/redux/book/actions.ts (hunk 2 above)
Comment (ghost) [reply wbooks/src/redux/book/actions.ts]: Done!

File: wbooks/src/services/BookService.ts (hunk 3 above)
Comment (ghost) [reply wbooks/src/services/BookService.ts]: Done!

File: wbooks/src/app/components/FilterInput/index.tsx (hunk 1 above)
Comment (ghost) [reply wbooks/src/app/components/FilterInput/index.tsx]: Done!

File: wbooks/src/redux/book/reducer.ts [hunk 4]
```diff
@@ -3,12 +3,24 @@ import { BookState } from '@interfaces/Book';
...
 
 const bookReducer = {
-  primaryActions: [actions.GET_BOOKS, actions.GET_FILTERED_BOOKS]
+  primaryActions: [actions.GET_BOOKS],
+  override: {
+    [actions.SET_FILTER_QUERY]: (state: BookState, action: Action) => ({
+      ...state,
+      [action.target]: action.payload
+    })
+  }
```
Comment (sfernandez11) [inline wbooks/src/redux/book/reducer.ts]: Check this https://github.com/Wolox/redux-recompose/tree/master/src/effects/onReadValue

Comment (ghost) [reply wbooks/src/redux/book/reducer.ts]: o.O Done!

**Review by sfernandez11 — DISMISSED** [review]
🥇

**Review by sfernandez11 — APPROVED** [review]

**Pull request merged**